osquery/tables/events/linux/bpf_file_events.cpp

```diff
@@ -0,0 +1,179 @@
+/**
+ * Copyright (c) 2014-present, The osquery authors
+ *
+ * This source code is licensed as defined by the LICENSE file found in the
+ * root directory of this source tree.
+ *
+ * SPDX-License-Identifier: (Apache-2.0 OR GPL-2.0-only)
+ */
+
+#include <osquery/logger/logger.h>
+#include <osquery/config/config.h>
+#include <osquery/registry/registry_factory.h>
+#include <osquery/sql/sql.h>
+#include <osquery/tables/events/linux/bpf_file_events.h>
+#include <osquery/filesystem/filesystem.h>
+#include <osquery/utils/system/uptime.h>
+
+#include <boost/algorithm/string.hpp>
+#include <iostream>
+#include <regex>
+
+namespace osquery {
+
+REGISTER(BPFFileEventSubscriber, "event_subscriber", "bpf_file_events");
+
+std::vector<std::string> monitored_paths;
+
+void configure_filtering() {
+    monitored_paths.clear();
+
+    VLOG(1) << "Configured BPF file event subscriber";
+    auto parser = Config::getParser("file_paths");
+    if (parser == nullptr) {
+        LOG(ERROR) << "No file paths found";
+        return;
+    }
+    auto const& doc = parser->getData().doc();
+    auto file_accesses_it = doc.FindMember("file_accesses");
+    if (file_accesses_it == doc.MemberEnd()) {
+    LOG(ERROR) << "No key 'file_accesses' found when parsing file events"
+                    " subscriber configuration.";
+    return;
+    }
+    auto& accesses = file_accesses_it->value;
+    if (accesses.GetType() != rapidjson::kArrayType) {
+    LOG(ERROR) << "Wrong type found for file_accesses when parsing file events"
+                    " subscriber configuration. Found ("
+                << accesses.GetType()
+                << "),"
+                    " expected array ("
+                << rapidjson::kArrayType << ").";
+    return;
+    }
+
+    Config::get().files([&accesses](const std::string& category,
+                const std::vector<std::string>& files) {
+                for (auto file : files) {
+                    replaceGlobWildcards(file);
+                    //std::cout << "File pattern: " << file << std::endl;
+
+                    std::vector<std::string> file_path_list = {};
+                    resolveFilePattern(file, file_path_list);
+
+                    //for (const auto& file_path : file_path_list) {
+                    //    std::cout << "\t File path: " << file_path << std::endl;
+                    //}
+
+                    monitored_paths.insert(monitored_paths.end(), file_path_list.begin(), file_path_list.end());
+                }
+    });
+}
+
+void BPFFileEventSubscriber::configure() {
+    configure_filtering();
+}
+
+Status BPFFileEventSubscriber::init() {
```

Comment:
Add a flag to enabled/disable this feature; only perform initialization when the flag is  enabled